src/pages/Search/SearchAdvancedFiltersPage/SearchFiltersCardPage.tsx

```diff
@@ -1,84 +1,262 @@
-import type {CompanyCardFeed} from '@src/types/onyx';
+import type {Card, CardList, CompanyCardFeed, WorkspaceCardsList} from '@src/types/onyx';
+import type {BankIcon} from '@src/types/onyx/Bank';
+import {isEmptyObject} from '@src/types/utils/EmptyObject';
+
+type CardFilterItem = Partial<OptionData> & {bankIcon?: BankIcon; lastFourPAN?: string; isVirtual?: boolean; isCardFeed?: boolean; correspondingCards?: string[]};
+
+type DomainFeedData = {bank: string; domainName: string; correspondingCardIDs: string[]};
+
+function getRepeatingBanks(workspaceCardFeedsKeys: string[], domainFeedsData: Record<string, DomainFeedData>) {
+    const bankFrequency: Record<string, number> = {};
+    for (const key of workspaceCardFeedsKeys) {
+        // Example: "cards_18755165_Expensify Card" -> "Expensify Card"
+        const bankName = key.split('_').at(2);
+        if (bankName) {
+            bankFrequency[bankName] = (bankFrequency[bankName] || 0) + 1;
+        }
+    }
+    for (const domainFeed of Object.values(domainFeedsData)) {
+        bankFrequency[domainFeed.bank] = (bankFrequency[domainFeed.bank] || 0) + 1;
+    }
+    return Object.keys(bankFrequency).filter((bank) => bankFrequency[bank] > 1);
+}
+
+function createIndividualCardFilterItem(card: Card, selectedCards: string[], iconStyles: StyleProp<ViewStyle>) {
+    const isSelected = selectedCards.includes(card.cardID.toString());
+    const icon = CardUtils.getCardFeedIcon(card?.bank as CompanyCardFeed);
+    const cardName = card?.nameValuePairs?.cardTitle ?? card?.cardName;
+    const text = card.bank === CONST.EXPENSIFY_CARD.BANK ? card.bank : cardName;
+
+    return {
+        lastFourPAN: card.lastFourPAN,
+        isVirtual: card?.nameValuePairs?.isVirtual,
+        text,
+        keyForList: card.cardID.toString(),
+        isSelected,
+        bankIcon: {
+            icon,
+            iconWidth: variables.cardIconWidth,
+            iconHeight: variables.cardIconHeight,
+            iconStyles,
+        },
+        isCardFeed: false,
+    };
+}
+
+function buildIndividualCardsData(
+    workspaceCardFeeds: Record<string, WorkspaceCardsList | undefined>,
+    userCardList: CardList,
+    selectedCards: string[],
+    iconStyles: StyleProp<ViewStyle>,
+): CardFilterItem[] {
+    const userAssignedCards: CardFilterItem[] = Object.values(userCardList ?? {}).map((card) => createIndividualCardFilterItem(card, selectedCards, iconStyles));
+
+    // When user is admin of a workspace he sees all the cards of workspace under cards_ Onyx key
+    const allWorkspaceCards: CardFilterItem[] = Object.values(workspaceCardFeeds)
+        .filter((cardFeed) => !isEmptyObject(cardFeed))
+        .flatMap((cardFeed) => {
+            return Object.values(cardFeed as Record<string, Card>)
+                .filter((card) => card && CardUtils.isCard(card) && !userCardList?.[card.cardID])
+                .map((card) => createIndividualCardFilterItem(card, selectedCards, iconStyles));
+        });
```

Comment:
Hmm yeah, this makes me wonder if we should reconsider the avatars we use for cards... should we consider something more like this? 
![CleanShot 2024-12-19 at 15 10 26@2x](https://github.com/user-attachments/assets/e201b206-097b-47ab-8b73-234d401fe883)

Reply:
Ooo I could definitely be into something like that.

Reply:
Can you link the whole figma for that design? Also does that mean that we should grup cards in Card Filter by user?

Reply:
My thinking is that this would work much like Workspace avatars do...

The workspace itself just uses a single avatar. In this case, the feed itself uses a single avatar and that's the big card logo. 

But then individual users/cards within the feed get the stacked avatar + card treatment like I'm showing above. 

Before we commit to that, let's see what @JmillsExpensify @trjExpensify @Expensify/design @joekaufmanexpensify thinks though. I think it could help solve this issue though.

Reply:
I think that's a cool idea, should we put your own assigned cards at the top of the "Individual cards" list anyway though as well?

Reply:
Agreed! I really like that pattern for allowing an admin to see to whom the card is assigned when filtering.